src/kubectlUtils.ts

```diff
@@ -367,3 +367,33 @@ export async function getResourceAsJson<T extends KubernetesResource | Kubernete
     }
     return shellResult.result;
 }
+
+export async function createResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    await changeResourceFromUri(uri, kubectl, 'create', 'creating', 'created');
+}
+
+export async function deleteResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    const result = await vscode.window.showWarningMessage('Are you sure you want to delete this resource?', 'Delete', 'Cancel');
+    if (result === 'Delete') {
+        await changeResourceFromUri(uri, kubectl, 'delete', 'deleting', 'deleted');
+    }
+}
+
+export async function applyResourceFromUri(uri: vscode.Uri, kubectl: Kubectl) {
+    await changeResourceFromUri(uri, kubectl, 'apply', 'applying', 'applied');
+}
+
+async function changeResourceFromUri(uri: vscode.Uri, kubectl: Kubectl, command: string, verbParticiple: string, verbPast: string) {
+    if (uri.scheme !== 'file') {
+        vscode.window.showErrorMessage(`${uri.toString()} is not a file path.`);
+        return;
+    }
+    const path = vscode.workspace.asRelativePath(uri);
+    const result = await kubectl.invokeAsync(`${command} -f '${path}'`);

```

Comment:
I'm not sure single quotes will work on Windows.  Single quotes seem to be interpreted by the Windows shell as filename characters, and if I run `kubectl delete -f 'some_file_that_exists.yaml'` at the DOS prompt then I get a 'path does not exist' error.  Is it working for you in non-WSL Windows?  If not I think double quotes work on all OSes.

Reply:
done.